### PR TITLE
Issue 155

### PR DESCRIPTION
**Pull request by Kourchenko**
UI fix for requesting location from a user.
Added a new form '_form_step_0.html.erb' that either requests location on 'Enable Location', takes an Address as input (using Algolia), or does not track their location. We only track it if the user entered in a full address or has already enabled location.

Simplified the '_error_geolocation.html.erb' page to show basic information about locaiton.

### CODE REVIEW

File: app/views/home/_form_step_0.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  <div class='checkboxes-container'>
+    <p class='bold'>To put your speed test results on the map, we'd like to use your location.</p>
+    <div class='checkbox checkbox-info'>
+      <%= f.check_box :location, { id: 'location_geolocation', data: { target: '#form-step-1' } }, 'Location Geolocation', 'true' %>
+      <%= label_tag :location_geolocation, "Enable Location".html_safe %>
+    </div>
+
+    <div class='checkbox checkbox-info'>
+      <%= f.check_box :location, { id: 'location_address', data: { target: '#form-step-1' } }, 'Location Address', false %>
```
Comment (Kourchenko) [inline app/views/home/_form_step_0.html.erb]: I really don't need a ":location" here, but Ruby on Rails throws an error.
If the reviewer has input on how to resolve this, I would be much appreciative.

File: app/assets/javascripts/home.coffee [hunk 2]
```diff
@@ -117,15 +145,30 @@ set_error_for_invalid_fields = ->
       $('#submission_provider_down_speed').removeClass('got-error')
       $('#speed_error_span').addClass('hide')
 
+places_autocomplete = ->
+
+  placesAutocomplete = places({
+    application_id: 'pl1SUFESCKRV',
+    api_key: '6039fe8e924c5e9f9a2edd9cecba075c',
```
Comment (ryanrolds) [inline app/assets/javascripts/home.coffee]: Is this API key sensitive?

File: app/assets/javascripts/home.coffee [hunk 3]
```diff
@@ -38,7 +43,30 @@ set_coords = (position) ->
         Sentry.setExtra("response_status",  statusText)
         Sentry.setExtra("response_error",  errorText)
         Sentry.captureException(err)
-      
+
+set_coords_by_latlng = (latlng) ->
+  $('submission_latitude').attr 'value', latlng.lat
+  $('submission_latitude').attr 'value', latlng.lng
+  $.ajax
```
Comment (ryanrolds) [inline app/assets/javascripts/home.coffee]: A lot of this function is similar to set_cords. Could they be merged and the success handler vary for each context?

**Comment by ryanrolds** [comment]
It's going in the right direction.

There are styling differences between the buttons and null space around the buttons. I think the text of the selections needs to be improved, they look like they are written by an engineer and are not user-friendly. Buttons that are not clickable do not look disabled. Should the browser location option be already selected? Should the get location button change if the location is enabled or after the user clicks it? 

Overall the logic works, I think it needs some UX focused adjustments.

File: app/assets/javascripts/home.coffee (hunk 2 above)
Comment (Kourchenko) [reply app/assets/javascripts/home.coffee]: API Key is a public key. According to their documentation:
> This is the public API key to use in your frontend code. This key is only usable for search queries.
I think we should store it elsewhere and reference it. Not sure how to do that.

Comment (ryanrolds) [reply app/assets/javascripts/home.coffee]: Thank you.

File: app/assets/javascripts/home.coffee [hunk 4]
```diff
@@ -15,22 +17,32 @@ bind_rating_stars = ->
 disable_form_inputs = ->
   $('#form-container .form-fields input').prop('disabled', true)
 
+<<<<<<< HEAD
```
Comment (ryanrolds) [inline app/assets/javascripts/home.coffee]: A merge wasn't resolved and both copies were checked in.

**Comment by ryanrolds** [comment]
Now that the accuracy is merged, what will the accuracy for an address be?

**Comment by Kourchenko** [comment]
> 
> 
> Now that the accuracy is merged, what will the accuracy for an address be?

We could operate on the idea that a user entering in an address is their location, and would thus be the most accurate measurement of 0 meters. What happens if a user enters in an address where they don't live? In such a case our accuracy and that test would be inaccurate.

**Pull request merged**